core/iwasm/compilation/aot_emit_aot_file.c

```diff
@@ -1506,31 +1505,150 @@ aot_emit_table_info(uint8 *buf, uint8 *buf_end, uint32 *p_offset,
+            if (types[i]->type_flag == DEFINED_TYPE_FUNC) {
+                AOTFuncType *func_type = (AOTFuncType *)types[i];
+                EMIT_U16(func_type->param_count);
+                EMIT_U16(func_type->result_count);
+                EMIT_U16(func_type->param_cell_num);
+                EMIT_U16(func_type->ret_cell_num);
+                EMIT_U16(func_type->ref_type_map_count);
+                EMIT_BUF(func_type->types,
+                         func_type->param_count * func_type->result_count);
+            }
+            /* Emit DEFINED_TYPE_STRUCT */
+            else if (types[i]->type_flag == DEFINED_TYPE_STRUCT) {
+                AOTStructType *struct_type = (AOTStructType *)types[i];
+                EMIT_U32(struct_type->total_size);
+                EMIT_U16(struct_type->field_count);
+                EMIT_U16(struct_type->ref_type_map_count);
+                for (idx = 0; idx < struct_type->field_count; idx++) {
+                    EMIT_U16(struct_type->fields[idx].field_flags);
+                    EMIT_U8(struct_type->fields[idx].field_type);
+                    EMIT_U8(struct_type->fields[idx].field_size);
+                    EMIT_U32(struct_type->fields[idx].field_offset);
+                }
+                aot_emit_reftype_map(buf, buf_end, &offset,
+                                     struct_type->ref_type_map_count,
+                                     struct_type->ref_type_maps);
+            }
+            /* Emit DEFINED_TYPE_ARRAY */
+            else if (types[i]->type_flag == DEFINED_TYPE_ARRAY) {
+                AOTArrayType *array_type = (AOTArrayType *)types[i];
+                EMIT_U16(array_type->elem_flags);
+                EMIT_U8(array_type->elem_type);
+                EMIT_U8(array_type->elem_ref_type->ref_type);
+                if (wasm_is_reftype_htref_nullable(
+                        array_type->elem_ref_type->ref_type)
+                    || wasm_is_reftype_htref_non_nullable(
+                        array_type->elem_ref_type->ref_type)) {
+                    EMIT_U16(array_type->elem_ref_type->ref_ht_common.nullable);
+                    EMIT_U32(
+                        array_type->elem_ref_type->ref_ht_common.heap_type);
+                }
+            }
+            else {
+                aot_set_last_error("invalid type flag.");
+                return false;
+            }
+        }
+
+        if (offset - *p_offset
+            != get_func_type_info_size(comp_ctx, comp_data)) {
+            aot_set_last_error("emit function type info failed.");
+            return false;
+        }
+
+        *p_offset = offset;
     }
+    else
+#endif
+    {
+        AOTFuncType **func_types = (AOTFuncType **)comp_data->types;
 
-    *p_offset = offset;
+        for (i = 0; i < comp_data->type_count; i++) {
+            offset = align_uint(offset, 4);
+            /* If GC enabled, only emit function type info */
+            EMIT_U32(DEFINED_TYPE_FUNC);
```

Comment:
To keep consistent with GC AOT format, here we should also emit is_sub_final and parent_type_idx, though they are ignored?
This makes that the non-GC AOT file can be loaded by GC/non-GC AOT runtime.